lib/svgo/coa.js

```diff
@@ -215,7 +215,7 @@ async function action(args, opts, command) {
   // --output
   if (output) {
     if (input.length && input[0] != '-') {
-      if (output.length == 1 && checkIsDir(output[0])) {
+      if (output.length == 1 && output[0] != '-' && !path.extname(output[0])) {
```

Comment:
I disagree with this approach because it leads to unexpected behavior in other cases, for example:

```sh
svgo original.svg -o minified
```

I would expect this to create a file, not a directory, called `minified`.

Imo, a better solution would be to update `checkIsDir` to return `true` when the path ends with a trailing `path.sep`?

```js
/**
 * Synchronously check if path is a directory. Tolerant to errors like ENOENT.
 *
 * @param {string} filePath
 */
export function checkIsDir(filePath) {
  try {
    return fs.lstatSync(filePath).isDirectory();
  } catch (e) {
    return filePath.endsWith(path.sep);
  }
}
```

```sh
# output: minified
svgo original.svg -o minified

# output: minified.svg
svgo original.svg -o minified.svg

# output: dist/minified.svg
svgo original.svg -o dist/minified.svg

# output: dist/original.svg
svgo original.svg -o dist/
```

Now, when the output path ends with an explicit `/` (or `\` on Windows) we know you mean a directory. Backward compatibility is less of a concern because SVGO previously crashed in this scenario anyway. (When the path ends with a trailing slash.)

Does this sound like a suitable solution to you?